crapssim/bet.py

```diff
@@ -89,40 +58,100 @@ def allowed(self, table: 'Table') -> bool:
         """
         return True
 
+    @property
+    @abstractmethod
+    def status(self) -> str | bool:
+        pass
+
+    @property
+    def win_amount(self) -> float:
+        if self.status == "win":
+            return self.payout_ratio * self.bet_amount
+        return 0.0
+
+    @property
+    def remove(self) -> bool:
+        if self.status is not None:
+            return True
+        return False
+
+    def _update_bet(self) -> tuple[str | None, float, bool]:
+        """
+        Returns whether the bets status is win, lose or None and if win the amount won.
+
+        Returns
+        -------
+        tuple[str | None, float]
+            The status of the bet and the amount of the winnings.
+        """
+        return self.status, self.win_amount, self.remove
+
+
+class WinningLosingNumbersBet(Bet, ABC):

```

Comment:
Since so many bets have this as a parent (PassLine, DontPass, Come, DontCome, Place, Yo/Boxcars/Any7/etc., Field, plus Buy/Lay bets in the future), can we call it something more generic?

Maybe SimpleBet or StandardBet?  

Really the only bets that don't follow this structure are ATS, Fire, and Hardways

Reply:
Maybe slight preference to SimpleBet, but both work for me